ui/app/src/modules/Preview/Tools/BrowseComponentsPanel.tsx

```diff
@@ -0,0 +1,328 @@
+    errorSelector: source => source.error,
+    resultSelector: source => {
+
+      const items = source.page[source.pageNumber]?.map((id) => source.byId[id]) || [];
+      return {
+        ...pluckProps(source, 'count', 'query.limit', 'pageNumber', 'contentTypeFilter'),
+        items
+      } as ComponentResource
+    }
+  });
+  const { formatMessage } = useIntl();
+  const AUTHORING_BASE = useSelection<string>(state => state.env.AUTHORING_BASE);
+  const hostToGuest$ = getHostToGuestBus();
+
+  const onDragStart = (item: ContentInstance) => hostToGuest$.next({
+    type: COMPONENT_INSTANCE_DRAG_STARTED,
+    payload: item
+  });
+
+  const onDragEnd = () => hostToGuest$.next({ type: COMPONENT_INSTANCE_DRAG_ENDED });
+
+  useEffect(() => {
+    dispatch(fetchComponentsByContentType(contentTypeFilter));
+  }, [contentTypeFilter, dispatch]);
+
+  useEffect(() => {
+    const subscription = onSearch$.pipe(
+      debounceTime(400),
+      distinctUntilChanged()
+    ).subscribe((keywords: string) => {
+      dispatch(fetchComponentsByContentType(null, { keywords }));
+    });
+    return () => subscription.unsubscribe();
+  }, [dispatch, onSearch$]);
+
+  function onPageChanged(event: React.MouseEvent<HTMLButtonElement, MouseEvent> | null, newPage: number) {
+    dispatch(fetchComponentsByContentType(null, { offset: newPage }));
+  }
+
+  function handleSearchKeyword(keyword: string) {
+    setKeyword(keyword);
+    onSearch$.next(keyword);
+  }
+
+  function handleSelectChange(value: string) {
+    setContentTypeFilter(value);
+  }
+
+  return (
+    <ToolPanel title={translations.browse}>
+      <ErrorBoundary>
+        <div className={classes.search}>
+          <SearchBar
+            onChange={handleSearchKeyword}
+            keyword={keyword}
+            disabled={isFetching}
+          />
+          {
+            contentTypes &&
+            <Select
+              value={contentTypeFilter}
+              displayEmpty
+              className={classes.Select}
+              onChange={(event: any) => handleSelectChange(event.target.value)}
+              disabled={isFetching}
+            >
+              <MenuItem value="" disabled>{formatMessage(translations.selectContentType)}</MenuItem>
+              {
+                contentTypes.map((contentType: ContentType, i: number) => {
+                  return <MenuItem value={contentType.id} key={i}>{contentType.name}</MenuItem>
+                })
+              }
+            </Select>
+          }
+        </div>
+        <React.Suspense
+          fallback={
+            <LoadingState
+              title={formatMessage(translations.loading)}
+              graphicProps={{ width: 150 }}
+            />
+          }
+        >
+          <BrowsePanelUI
+            componentsResource={resource}
+            classes={classes}
+            onPageChanged={onPageChanged}
+            onDragStart={onDragStart}
+            onDragEnd={onDragEnd}
+            AUTHORING_BASE={AUTHORING_BASE}
```

Comment:
Looks awkward a prop like this... let's just use camelcase: `authoringBase={AUTHORING_BASE}`